components/CampaignCard.jsx

```diff
@@ -0,0 +1,48 @@
+import React, { useEffect, useState } from 'react';
+import { faDonate } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+const CampaignCard = ({ name, text, logoUrl, donate }) => {
+    const [state, setstate] = useState(false);
+    const [details, setDetails] = useState('');
```

Comment:
Keep only one state
Eg:
`const [isReadMore, setIsReadMore] = useState(false);`
and conditionally render text 📌 like
`isReadMore ?  text : text.substr(0, 300)`
so that you can also remove `useEffect` 📍